include/dts-functions.sh

```diff
@@ -1730,72 +1623,90 @@ send_dts_logs(){
 }
 
 check_if_fused() {
-  file_path="/sys/class/mei/mei0/fw_status"
-
-  if [[ ! -f $file_path ]]; then
-    echo "File not found: $file_path"
+  local _file_path
+  _file_path="/sys/class/mei/mei0/fw_status"
+  local _file_content
+  local _hfsts6_value
+  local _line_number
+  local _hfsts6_binary
+  local _binary_length
+  local _padding
+  local _zeros
+  local _bit_30_value
+
+  if ! $FSREAD_TOOL test -f "$_file_path"; then
+    print_error "File not found: $_file_path"
     return 2
   fi
 
-  hfsts6_value=""
-  line_number=1
+  _file_content="$($FSREAD_TOOL cat $_file_path)"
+
+  _fsts6_value=""
+  _line_number=1
   while IFS= read -r line; do
-    if [[ $line_number -eq 6 ]]; then
-      hfsts6_value=$line
+    if [[ $_line_number -eq 6 ]]; then
+      _hfsts6_value="$line"
       break
     fi
-    ((line_number++))
-  done <"$file_path"
+    ((_line_number++))
+  done <<< "$_file_content"
 
-  if [[ -z $hfsts6_value ]]; then
-    echo "Failed to read HFSTS6 value"
+  if [[ -z "$_hfsts6_value" ]]; then
+    print_error "Failed to read HFSTS6 value"
     exit 1
   fi
 
-  hfsts6_binary=$(echo "ibase=16; obase=2; $hfsts6_value" | bc)
-
-  binary_length=${#hfsts6_binary}
+  _hfsts6_binary=$(echo "ibase=16; obase=2; $_hfsts6_value" | bc)
+  _binary_length=${#_hfsts6_binary}
 
   # Add leading zeros
-  if [ $binary_length -lt 32 ]; then
-    padding=$((32 - $binary_length))
-    zeros=$(printf "%${padding}s" | tr ' ' "0")
-    hfsts6_binary=$zeros$hfsts6_binary
+  if [ $_binary_length -lt 32 ]; then
+    _padding=$((32 - $_binary_length))
+    _zeros=$(printf "%${_padding}s" | tr ' ' "0")
+    _hfsts6_binary=$_zeros$_hfsts6_binary
   fi
 
-  bit_30_value=${hfsts6_binary:1:1}
+  _bit_30_value=${_hfsts6_binary:1:1}
 
-  if [ $bit_30_value == 0 ]; then
+  if [ $_bit_30_value == 0 ]; then
     return 1
   else
     return 0
   fi
 }
 
 check_if_boot_guard_enabled() {
+  local _msr_hex
+  local _msr_binary
+  local _binary_length
+  local _padding
+  local _zeros
+  local _facb_fpf
+  local _verified_boot
+
   # MSR cannot be read
-  if ! rdmsr 0x13a -0; then
+  if ! $RDMSR 0x13a -0; then
     return 1
   fi
 
-  msr_hex=$(rdmsr 0x13a -0 | tr '[:lower:]' '[:upper:]')
```

Comment:
@PLangowski, you are checking for a value of a specific register here, am I right? If so, is the layout of these registers is the same for all platforms or MEI's? Could you point me out to the documentation where this register is described? I want to add some comments here, because the purpose and the method of this functions is not clear.

Reply:
I wish I could help with the documentation, but I don't  think it can be found https://github.com/Dasharo/dts-scripts/pull/15#discussion_r1793381234

Reply:
Will be addressed here https://github.com/Dasharo/dts-scripts/pull/37#pullrequestreview-2381558744.